lib/config/index.js

```diff
@@ -154,7 +154,6 @@ function Config(config) {
   this.browser_monitoring.loader_version = ''
 
   // Settings to play nice with DLPs (see NODE-1044).
-  this.compressed_content_encoding = 'deflate' // Deflate or gzip
```

Comment:
Don't we still need this setting here? I see that we have it defaulted in `lib/config/default.js`, but this change is inside a catch statement for if we fail to pull the default config, so don't we still need it?

Reply:
I'm trying to figure out what could happen.

If we somehow don't get the value from the defaults (not sure how this could happen, the try-catch above to check this seems overly cautious), then it won't be set either from user config further below in `this._fromPassed(config)`. That means that when it's time to check the value of `compressed_content_encoding`, we're going to get `undefined`, which isn't equal to `'gzip'` in the test here:

https://github.com/newrelic/node-newrelic/blob/main/lib/collector/remote-method.js#L155

so the user will get deflate compression instead of the supposed default.

Anyway, I think it's okay to remove setting this config here, since these are supposed to be only internal, non-configurable configs (?), which this one isn't anymore. But I do think I should change the default in case something goes wrong with the config and we don't default to deflate.

Reply:
It just seems like we could have `this.compressed_content_encoding = 'gzip'` here, and that would take care of the situation that you're describing without having to mess with the logic in `remote-method.js`. Either way, we're on the same page about making sure we handle that case one way or another :)

Reply:
I just checked and [one of the tests](https://github.com/newrelic/node-newrelic/blob/main/test/unit/collector/remote-method.test.js#L195) uses `BARE_AGENT` which has a completely empty config and ignores all logic on how to set config variables, including the change we're discussing. This was defaulting to `deflate`, changing the compression sizes, so I think the change in `remote-method.js` is needed after all.